### PR TITLE
 Fix serial wifi and bwifi. Crash bwifi when no ...

### PR DESCRIPTION
**Pull request by unlogisch04**
password is provided and add in wifi and bwifi length check of the ssid and password.
Fix:https://github.com/SlimeVR/SlimeVR-Tracker-ESP/issues/292

### CODE REVIEW

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: src/serial/serialcommands.cpp [hunk 1]
```diff
@@ -40,28 +40,82 @@ namespace SerialCommands {
...
-					WiFiNetwork::setWiFiCredentials(parser->getCmdParam(2), parser->getCmdParam(3));
+					const char *sc_ssid = parser->getCmdParam(2);
+					const char *sc_pw = parser->getCmdParam(3);
+
+					if (!lengthCheck(sc_ssid, 32, "CMD SET WIFI", "SSID")) {
+						return;
+					}
+					if (!lengthCheck(sc_pw, 64, "CMD SET WIFI", "Password")) {
+						return;
+					}
```
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp:72]: I do overriead the return so far back in a line. So it helps me to read/understand the code. Also the code was multiline when i did first begin.
Also: https://discord.com/channels/817184208525983775/932240516656029746/1154968426603888751

[the remark this answers is not in the document]
Comment (TheDevMinerTV) [reply src/serial/serialcommands.cpp:69]: you could've atleast put both checks into the same if

File: src/serial/serialcommands.cpp [hunk 2]
```diff
@@ -40,28 +40,82 @@ namespace SerialCommands {
...
-					WiFiNetwork::setWiFiCredentials(ssid, pass);
+
+                    if (b64ssid!=NULL) {
+			            b64ssidlength = strlen(b64ssid);
+						ssidlength = decode_base64_length((unsigned char *)b64ssid, (unsigned int)b64ssidlength);
+		            }
+                    if (b64pass!=NULL) {
+			            b64passlength = strlen(b64pass);
+						passlength = decode_base64_length((unsigned char *)b64pass, (unsigned int)b64passlength);
+		            }
```
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp:93]: I did add some. But it was not possible all the way. A other suggestion?

File: src/serial/serialcommands.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp:69]: done

File: src/serial/serialcommands.h [hunk 3]
```diff
@@ -25,9 +25,23 @@
...
+ *     If (the pointer is Null) and (the text is smaller than length) it returns true.
+ *   Parameters:
+ *     text - Pointer to input const char*
+ *     length - How long the text max should be
+ *     cmd - Begin of the error text
+ *     name - How the text should be Called in the error Message
+ *   Returns:
+ *     true when size is smaller or equal than lenght or pointer = NULL
+ *     false when size is bigger than length error "%s ERROR: %s is longer than %d bytes / Characters"
+ */
```
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.h]: i removed it as i not intended to use that outside

File: src/serial/serialcommands.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp]: done

File: src/serial/serialcommands.cpp [hunk 4]
```diff
@@ -196,7 +255,9 @@ namespace SerialCommands {
 			}
 
 			// Restore conencting state
-			if (WiFi.status() != WL_CONNECTED) WiFi.begin();
+			if (WiFi.status() != WL_CONNECTED) {
+				WiFi.begin();
+			}
```
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp]: reverted as i did not touch that lines.

File: src/serial/serialcommands.cpp [hunk 5]
```diff
@@ -177,7 +231,12 @@ namespace SerialCommands {
 			logger.info("[WSCAN] Scanning for WiFi networks...");
 
 			// Scan would fail if connecting, stop connecting before scan
-			if (WiFi.status() != WL_CONNECTED) WiFi.disconnect();
+			if (WiFi.status() != WL_CONNECTED) {
+				WiFi.disconnect();
+			}
```
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp]: reverted as i did not touch that lines.

File: src/serial/serialcommands.cpp [hunk 6]
```diff
@@ -40,28 +40,82 @@ namespace SerialCommands {
...
+					decode_base64((const unsigned char *)b64ssid, (unsigned int)b64ssidlength, (unsigned char*)ssid);
+					if (!lengthCheck(ssid, 32, "CMD SET BWIFI", "SSID")) {
+						return;
+					}
+
+					if ((b64pass!=NULL) && (b64passlength > 0)) {
+						decode_base64((const unsigned char *)b64pass, (unsigned int)b64passlength, (unsigned char*)pass);
+						if (!lengthCheck(ssid, 64, "CMD SET BWIFI", "Password")) {
+							return;
+						}
```
[the remark this answers is not in the document]
Comment (unlogisch04) [reply src/serial/serialcommands.cpp]: Will not change.

File: src/serial/serialcommands.h [hunk 7]
```diff
@@ -25,9 +25,9 @@
 #define SLIMEVR_SERIALCOMMANDS_H_
 
 namespace SerialCommands {
-    void setUp();
+	void setUp();
```
Comment (TheDevMinerTV) [inline src/serial/serialcommands.h:28]:
```suggestion
    void setUp();
```

**Review by TheDevMinerTV — APPROVED** [review]
code looks fine to me

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**